packages/common/src/store/pages/chat/slice.ts

```diff
@@ -156,26 +156,61 @@ const slice = createSlice({
     goToChat: (_state, _action: PayloadAction<{ chatId?: string }>) => {
       // triggers saga
     },
+    fetchLatestChats: (state) => {
+      // triggers saga
+      state.chats.status = 'REFRESHING'
+    },
     fetchMoreChats: (state) => {
       // triggers saga
       state.chats.status = Status.LOADING
     },
     fetchMoreChatsSucceeded: (
       state,
-      action: PayloadAction<TypedCommsResponse<UserChat[]>>
+      action: PayloadAction<
+        Pick<TypedCommsResponse<UserChat[]>, 'data' | 'summary'>
+      >
     ) => {
+      const { data, summary } = action.payload
       state.chats.status = Status.SUCCESS
-      state.chats.summary = action.payload.summary
-      for (const chat of action.payload.data) {
+      if (!state.chats.summary) {
+        state.chats.summary = summary
+      } else {
+        if (
+          summary?.next_cursor &&
+          dayjs(summary?.next_cursor).isAfter(state.chats.summary.next_cursor)
+        ) {
+          state.chats.summary.next_cursor = summary?.next_cursor
+          state.chats.summary.next_count = summary?.next_count
+        }
+        if (
+          summary?.prev_cursor &&
+          dayjs(summary?.prev_cursor).isBefore(state.chats.summary.prev_cursor)
+        ) {
+          state.chats.summary.prev_cursor = summary?.prev_cursor
+          state.chats.summary.prev_count = summary?.prev_count
+        }
+      }
+      for (const chat of data) {
         if (!(chat.chat_id in state.messages)) {
           state.messages[chat.chat_id] = chatMessagesAdapter.getInitialState()
         }
       }
-      chatsAdapter.addMany(state.chats, action.payload.data)
+      chatsAdapter.upsertMany(state.chats, data)
     },
     fetchMoreChatsFailed: (state) => {
       state.chats.status = Status.ERROR
     },
+    fetchLatestMessages: (state, action: PayloadAction<{ chatId: string }>) => {
+      // triggers saga
+      if (!state.messages[action.payload.chatId]) {
```

Comment:
nit: could clean this up - destructure `chatId` and is there a way to initialize state using `entityAdapter` fns?

Reply:
Yep it's initted below with entityAdapter, but agree we can destructure